Azure AD entities description (#896)
microsoft/onefuzz: docs/AADEntitites.md

> # Azure Ad Entities

Reviewer: ```suggestion
# Azure Active Directory Entities
```

> This document describes the configuration of entities create in azure AD by our [deployment script](../src/deployment/deploy.sh)

Reviewer: ```suggestion
This document describes the configuration of entities create in Azure AD by our [deployment script](../src/deployment/deploy.sh)
```

> This document describes the configuration of entities create in azure AD by our [deployment script](../src/deployment/deploy.sh)

Reviewer: ```suggestion
This document describes the configuration of entities create in azure AD by our [deployment script](../src/deployment/deploy.py).
```

> This document describes the configuration of entities create in Azure AD by our [deployment script](../src/deployment/deploy.sh)

Author: ```suggestion
This document describes the configuration of entities create in Azure AD by our [deployment script](../src/deployment/deploy.py)
```

> ### Onefuzz Application registration

Reviewer: ```suggestion
### OneFuzz Application Registration
```

> This is the registration of the onefuzz instance

Reviewer: ```suggestion
This is the registration of the OneFuzz instance:
```

> This is the registration of the OneFuzz instance:

Reviewer: ```suggestion
This is the registration of the OneFuzz instance.
```

Reviewer: (I suggested `:` previously, but it actually looks weird in context, sorry about that!)

> * name : <instance_name>

Reviewer: ```suggestion
* name : `<instance_name>`
```

* app roles

>     * ManagedNode

Reviewer: ```suggestion
    * _ManagedNode_
```

        * value: ManagedNode
        * Allowed Member types: Applications

>     * CliClient

Reviewer: ```suggestion
    * _CliClient_
```

        * value: ManagedNode
        * Allowed Member types: Applications

> * Api Permissions

Reviewer: ```suggestion
* API Permissions
```

>     * User.Read (Microsoft Graph)

Reviewer: Suggestion: link to docs, if available.

Reviewer: (apply this comment to any other standard API permissions that are referenced)

Reviewer: Found the docs for this:
```suggestion
    * _User.Read_ ([Microsoft Graph](https://docs.microsoft.com/en-us/graph/permissions-reference#user-permissions))
```

* scope

>     * user_impersonation

Reviewer: ```suggestion
    * `user_impersonation`
```

> * Authorozed application:

Reviewer: ```suggestion
* Authorized application:
```

>     * Onefuzz cli registration

Reviewer: ```suggestion
    * OneFuzz CLI registration
```

### Onefuzz Application Service Principal

> service principal linked to the onefuzz application registration

Reviewer: ```suggestion
Service principal linked to the OneFuzz application registration:
```

> Service principal linked to the OneFuzz application registration:

Reviewer: ```suggestion
Service principal linked to the OneFuzz application registration.
```

> * name: <instance_name>

Reviewer: ```suggestion
* name: `<instance_name>`
```

> * Application Id: <Onefuzz Application registration app_id>

Reviewer: ```suggestion
* Application Id: <OneFuzz Application registration app_id>
```

> * Application Id: <OneFuzz Application registration app_id>

Reviewer: Without escaping or annotation, the angle-bracketed code gets treated as an (invalid) markup tag, and is absent when rendered:
```suggestion
* Application Id: _&lt;OneFuzz Application registration `app_id`&gt;_
```
There are a few similar spots below.

Reviewer 2: ```suggestion
* Application Id: `<OneFuzz Application registration app_id>`
```

> ### Onefuzz cli registration

Reviewer: ```suggestion
### OneFuzz CLI registration
```

> The registration for the Command line interface

Reviewer: ```suggestion
The registration for the command line interface:
```

> The registration for the command line interface:

Reviewer: ```suggestion
The registration for the command line interface.
```

> * name: <instance_name>-cli

Reviewer: ```suggestion
* name: `<instance_name>-cli`
```

> ### Onefuzz cli Service Principal

Reviewer: ```suggestion
### OneFuzz CLI Service Principal
```

> service principal linked to the onefuzz cli application registration

Reviewer: ```suggestion
Service principal linked to the OneFuzz CLI application registration:
```

> Service principal linked to the OneFuzz CLI application registration:

Reviewer: ```suggestion
Service principal linked to the OneFuzz CLI application registration.
```

> * name: <instance_name>-cli

Reviewer: ```suggestion
* name: `<instance_name>-cli`
```

> * Application Id: <Onefuzz cli registration app_id>

Reviewer: ```suggestion
* Application Id: <OneFuzz CLI registration app_id>
```

> * Application Id: <OneFuzz CLI registration app_id>

Reviewer: ```suggestion
* Application Id: _&lt;OneFuzz CLI registration `app_id`&gt;_
```

Reviewer 2: ```suggestion
* Application Id: `<OneFuzz CLI registration app_id>`
```

> * User Assignment required: true

Reviewer: ```suggestion
* User Assignment required: _true_
```

* Permission

>     * CliClient (from Onefuzz Application registration)

Reviewer: ```suggestion
    * CliClient (from OneFuzz Application registration)
```

>     * CliClient (from OneFuzz Application registration)

Reviewer: ```suggestion
    * _CliClient_ (from OneFuzz Application registration)
```

> ### Managed node Service Principal

Reviewer: ```suggestion
### Managed Node Service Principal
```

> This entity is available after the first deployment. This is the service principal associated with the user managed identity <instance_name>-scalesetid

Reviewer: ```suggestion
This entity is available after the first deployment. This is the service principal associated with the user-assigned managed identity `<instance_name>-scalesetid`.
```

> This entity is available after the first deployment. This is the service principal associated with the user-assigned managed identity `<instance_name>-scalesetid`.

Reviewer: To match the usage below:
```suggestion
This entity is available after the first deployment. This is the service principal associated with the user-assigned managed identity `<instance_name>-<scaleset_id>`.
```

> * name: <instance_name>-scalesetid

Reviewer: ```suggestion
* name: `<instance_name>-<scaleset_id>`
```

* Service Principal
    * Permission

>         * ManagedNode (from Onefuzz Application registration)

Reviewer: ```suggestion
        * ManagedNode (from OneFuzz Application registration)
```

>         * ManagedNode (from Onefuzz Application registration)

Author: ``` Suggestion
        * ManagedNode (from OneFuzz Application registration)
```

>         * ManagedNode (from OneFuzz Application registration)

Reviewer: ```suggestion
        * _ManagedNode_ (from OneFuzz Application registration)
```
